diff --git a/minui/graphics.cpp b/minui/graphics.cpp
index f3380aaa..db023656 100644
--- a/minui/graphics.cpp
+++ b/minui/graphics.cpp
@@ -735,14 +735,14 @@ int gr_init(void)
 #else
 	printf("Skipping adf graphics because TW_TARGET_USES_QCOM_BSP := true\n");
 #endif
-
+/*
     if (!gr_draw) {
         backend = std::make_unique<MinuiBackendDrm>();
         gr_draw = backend->Init();
         if (gr_draw)
             printf("Using drm graphics.\n");
     }
-
+*/
     if (!gr_draw) {
         backend = std::make_unique<MinuiBackendFbdev>();
         gr_draw = backend->Init();
