diff --git a/libpixelflinger/Android.mk b/libpixelflinger/Android.mk
index 026e8ad..8fd2709 100644
--- a/libpixelflinger/Android.mk
+++ b/libpixelflinger/Android.mk
@@ -1,4 +1,9 @@
 LOCAL_PATH:= system/core/libpixelflinger
+
+ifneq ($(filter x86%,$(TARGET_ARCH)),)
+include $(call all-named-subdir-makefiles,codeflinger/x86/libenc)
+endif
+
 include $(CLEAR_VARS)
 
 #
@@ -8,7 +13,7 @@ include $(CLEAR_VARS)
 include $(CLEAR_VARS)
 
 ifneq ($(wildcard system/core/libpixelflinger/codeflinger/x86/X86Assembler.cpp),)
-    ifeq ($(TARGET_ARCH),x86)
+    ifneq ($(filter x86%,$(TARGET_ARCH)),)
         TW_HAVE_X86_ACCELERATED_PIXELFLINGER := true
     endif
 endif
@@ -53,7 +58,7 @@ PIXELFLINGER_SRC_FILES_x86 := \
 	scanline.cpp
 
 PIXELFLINGER_C_INCLUDES_x86 := \
-	external/libenc
+	$(LOCAL_PATH)/codeflinger/x86/libenc
 
 endif
 
@@ -97,6 +102,7 @@ LOCAL_SRC_FILES := $(PIXELFLINGER_SRC_FILES)
 LOCAL_SRC_FILES_arm := $(PIXELFLINGER_SRC_FILES_arm)
 LOCAL_SRC_FILES_arm64 := $(PIXELFLINGER_SRC_FILES_arm64)
 LOCAL_SRC_FILES_x86 := $(PIXELFLINGER_SRC_FILES_x86)
+LOCAL_SRC_FILES_x86_64 := $(PIXELFLINGER_SRC_FILES_x86)
 LOCAL_SRC_FILES_mips := $(PIXELFLINGER_SRC_FILES_mips)
 ifneq ($(shell test $(PLATFORM_SDK_VERSION) -gt 20; echo $$?),0)
     LOCAL_SRC_FILES += $(LOCAL_SRC_FILES_$(TARGET_ARCH))
@@ -106,8 +112,9 @@ LOCAL_C_INCLUDES += $(LOCAL_EXPORT_C_INCLUDE_DIRS)
 LOCAL_CFLAGS := $(PIXELFLINGER_CFLAGS)
 LOCAL_CPPFLAGS := -Wno-unused-function
 LOCAL_C_INCLUDES_x86 := $(PIXELFLINGER_C_INCLUDES_x86)
+LOCAL_C_INCLUDES_x86_64 := $(PIXELFLINGER_C_INCLUDES_x86)
 ifeq ($(TW_HAVE_X86_ACCELERATED_PIXELFLINGER),true)
 LOCAL_WHOLE_STATIC_LIBRARIES += libenc
-LOCAL_C_INCLUDES += external/libenc
+LOCAL_C_INCLUDES += $(LOCAL_PATH)/codeflinger/x86/libenc
 endif
 include $(BUILD_STATIC_LIBRARY)
